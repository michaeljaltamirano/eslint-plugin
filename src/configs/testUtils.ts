import { execSync } from 'child_process';
import path from 'path';
import fs from 'fs';
import { isJson } from '../utils';

const getFilePath = (dirname: string) => (file: string) =>
	path.resolve(dirname, file);

// eslint-disable-next-line jest/no-export
export const generateTest = (
	dirname: string,
	config: Record<string, unknown>,
): void => {
	// eslint-disable-next-line jest/require-top-level-describe
	test('serializes to a snapshot', () => {
		const baseJson = JSON.stringify(config);
		const setPath = getFilePath(dirname);

		fs.writeFileSync(setPath('fixture.json'), baseJson, {
			flag: 'w',
		});

		const eslintConfig = execSync(
			`yarn run eslint --no-eslintrc -c ${setPath(
				'fixture.json',
			)} --print-config ${setPath('index.ts')}`,
			{
				encoding: 'utf8',
			},
		);

		const configJson: unknown = JSON.parse(
			eslintConfig.substring(eslintConfig.indexOf('{')),
		);

		if (!isJson(configJson)) {
			throw new Error('parse error');
		}

		expect(configJson).toMatchSnapshot();
	});
};
