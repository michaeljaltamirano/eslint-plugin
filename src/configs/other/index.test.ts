import { other } from '.';
import { generateTest } from '../testUtils';

describe('@michaeljaltamirano/eslint-plugin', () => {
	describe('other config', () => {
		generateTest(__dirname, other);
	});
});
