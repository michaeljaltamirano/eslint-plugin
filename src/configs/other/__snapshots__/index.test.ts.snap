// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`@michaeljaltamirano/eslint-plugin other config serializes to a snapshot 1`] = `
Object {
  "env": Object {},
  "globals": Object {},
  "ignorePatterns": Array [],
  "parserOptions": Object {},
  "plugins": Array [],
  "rules": Object {
    "arrow-parens": Array [
      "off",
    ],
    "no-console": Array [
      "error",
      Object {
        "allow": Array [
          "warn",
          "error",
        ],
      },
    ],
  },
  "settings": Object {},
}
`;
